lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -1637,6 +1637,11 @@ def cmd_set(*args)
       active_module.import_target_defaults
     end
 
+    # Warn the user that setting SSL may require changing RPORT
+    if name.upcase == 'SSL' && datastore[name] == true
+      print_warning('Setting the SSL option may require changing the RPORT value!')
+    end
```

Comment:
Maybe this should detect if it changes value? That way it catches the scenario of it being turned off:

```suggestion
    # Warn the user that changing the SSL value may require changing RPORT
    if name.casecmp('SSL') == 0 && datastore[name] != value
      print_warning('Changing the SSL option may require changing the RPORT value!')
    end
```

The detection logic would have to be further up though, as I think the value's already been set by this point

Reply:
Yeah, definitely. I was thinking the same, but I'll have to iterate on this after VMware is done. Thank you. :)

Reply:
I like this patch, so I'll commit it and then fix it up. Thanks!

Reply:
Since the value is already set by this point, as you indicated, I just saved the old value to be compared against.

Reply:
Reworded the descriptions a bit to clarify what should be clarified and simplify what everyone knows.